client/client.go

```diff
@@ -160,7 +161,19 @@ func NewMetadata(md map[string][]string) Metadata {
 func (m Metadata) Get(key string) []string {
 	vals := m.data[key]
 	if len(vals) == 0 {
-		return nil
+		// we didn't find the key, but perhaps it just has different cases?
```

Comment:
Have you explored the possibility of making keys lowercase on `NewMetadata`. I am not sure how this API is used, so there may be performance implications, but it would make the code much simpler

Reply:
Yes, but given this is on the hot path, I didn't want to incur extra processing when not absolutely needed. I expect the typical case to still hit the old code path, as users would likely use the same case as the header.